test/Npgsql.Tests/ReaderNewSchemaTests.cs

```diff
@@ -13,43 +14,37 @@ namespace Npgsql.Tests
     /// Note that this API is also available on .NET Framework.
     /// For the old DataTable-based API, see <see cref="ReaderOldSchemaTests"/>.
     /// </summary>
+    [TestFixture(false)] // Sync
+    [TestFixture(true)] // Async
     public class ReaderNewSchemaTests : TestBase
     {
         // ReSharper disable once InconsistentNaming
         [Test]
         public async Task AllowDBNull()
         {
-            using (var conn = await OpenConnectionAsync())
-            {
-                await using var _ = await CreateTempTable(conn, "nullable INTEGER, non_nullable INTEGER NOT NULL", out var table);
-
-                using (var cmd = new NpgsqlCommand($"SELECT nullable,non_nullable,8 FROM {table}", conn))
-                using (var reader = await cmd.ExecuteReaderAsync(CommandBehavior.SchemaOnly | CommandBehavior.KeyInfo))
-                {
-                    var columns = reader.GetColumnSchema();
-                    Assert.That(columns[0].AllowDBNull, Is.True);
-                    Assert.That(columns[1].AllowDBNull, Is.False);
-                    Assert.That(columns[2].AllowDBNull, Is.Null);
-                }
-            }
+            using var conn = await OpenConnectionAsync();
+            await using var _ = await CreateTempTable(conn, "nullable INTEGER, non_nullable INTEGER NOT NULL", out var table);
+
+            using var cmd = new NpgsqlCommand($"SELECT nullable,non_nullable,8 FROM {table}", conn);
+            using var reader = await cmd.ExecuteReaderAsync(CommandBehavior.SchemaOnly | CommandBehavior.KeyInfo);
+            var columns = async ? await reader.GetColumnSchemaAsync() : reader.GetColumnSchema();
```

Comment:
We could have a utility method on this class which executes either GetColumnSchemaAsync or GetColumnSchema based on the flag.